src/main/scala/com/databricks/spark/csv/package.scala

```diff
@@ -65,17 +84,35 @@ package object csv {
         "" // There is no need to generate header in this case
       }
       val strRDD = dataFrame.rdd.mapPartitions { iter =>
+        val csvFormatBase = CSVFormat.DEFAULT
+          .withDelimiter(delimiterChar)
+          .withEscape(escapeChar)
+          .withSkipHeaderRecord(false)
+          .withNullString("null")
+
+        val csvFormat = quote match {
+          case Some(s) => {
+            if (s.length == 1) {
```

Comment:
This check is better done outside of the loop, otherwise we will throw the exception on every element.